### PR TITLE
Ros2 devel

### PR DESCRIPTION
**Pull request by Karsten1987**
This is a continuation of #43 which hopefully some of the outstanding reviews addressed.

### CODE REVIEW

File: bondcpp/include/bondcpp/BondSM_sm.hpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
```
Comment (mjcarroll) [inline bondcpp/include/bondcpp/BondSM_sm.hpp]: I don't know if prepending the license like this is valid?

**Review by mjcarroll — CHANGES_REQUESTED** [review]
This looks like a straightforward port.  I know it's blocking for diagnostics, so it's probably okay to go out as is.

I do have some concern about prepending the apache license on some of these files, I'm not sure if that is valid behavior.

Also, potentially as an enhancement down the road, this should take advantage of lifecycle nodes, but that's probably not critical at this juncture.

**Comment by Karsten1987** [comment]
Oh yeah, I was raising my concerns about this as well but didn’t address it in this PR. I’ll revert them to their original license.

**Comment by Karsten1987** [comment]
https://github.com/ros/bond_core/pull/54/commits/71172695fe5c71063aaf5f05712545f7dab527d4 should remove the false declaration of Apache2.
I further disable the linter for copyright as a non-apache2 license currently doesn't match.

**Comment by Karsten1987** [comment]
@mjcarroll do you mind taking another look on this? 🙏

**Review by mjcarroll — APPROVED** [review]

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8105/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4104/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6605/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7986/)

**Comment by Karsten1987** [comment]
testing `smclib`, `bond`, `bondcpp`, `bond_core`, `test_bond`:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8106/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4105/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6606/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7987/)

**Pull request merged**